src/state/reducers/ShoppingCartReducer.js

```diff
@@ -0,0 +1,58 @@
+export const SEARCH_RESULTS_INITIAL_STATE = {
+    amountItems: 0,
+    items: []
+};
+
+export default function shoppingCartReducer(state = SEARCH_RESULTS_INITIAL_STATE, action) {
+    switch(action.type) {
+        case "add_cart":
+            //payload: { item: { id, qty }} 
+
+            if(state.items.some((item) => item.id === action.payload.id)) {
+                const elementIndex = state.items.findIndex((item) => item.id === action.payload.id);
+                let newArray = [...state.items];
+                newArray[elementIndex] = { ...newArray[elementIndex], 
+                    quantity: (newArray[elementIndex].quantity + action.payload.quantity <= newArray[elementIndex].stock)  
```

Comment:
it may be clearer if you move this little section to a variable and then only assign it here